airbyte-protocol/protocol-models/src/main/java/io/airbyte/protocol/models/CatalogHelpers.java

```diff
@@ -244,7 +244,20 @@ protected static List<Pair<List<String>, JsonNode>> getFullyQualifiedFieldNamesW
         .stream()
         // first node is the original object.
         .skip(1)
-        .filter(fieldWithSchema -> filterChildrenOfFoneOneOf(fieldWithSchema.getLeft(), fieldWithSchema.getRight(), fieldNamesThatAreOneOfs))
+        .filter(fieldWithSchema -> filterChildrenOfFoneOneOf(fieldWithSchema.getLeft(),
+            fieldWithSchema.getRight(),
+            fieldNamesThatAreOneOfs))
+        .map(fieldWithSchema -> {
+          if (withNamedOneOf && isOneOfField(fieldWithSchema.getRight())) {
+            final List<String> newPath = new ArrayList<>(fieldWithSchema.getLeft());
```

Comment:
Nit:  this could be reduced to a one liner:  `final List<String> newPath = List.of(fieldWithSchema.getLeft(), "oneOf");`

Reply:
I reverted the fix. We shouldn't allow it and have the format being enforced instead.